src/services/LedgerService.ts

```diff
@@ -0,0 +1,84 @@
+import { DerivationPathValidator } from '@/core/validation/validators';
+import { SymbolLedger } from '@/core/utils/Ledger';
+import TransportWebUSB from '@ledgerhq/hw-transport-webusb';
+const TransportNodeHid = window['TransportNodeHid'] && window['TransportNodeHid'].default;
+
+export class LedgerService {
+    private transport;
+
+    private async openTransport() {
+        return TransportNodeHid ? await TransportNodeHid.open() : await TransportWebUSB.create();
+    }
+
+    private async closeTransport() {
+        TransportNodeHid && this.transport && (await this.transport.close());
+    }
+
+    private formatError(error) {
+        return error.statusCode || error.id ? { errorCode: error.statusCode || error.id } : error;
+    }
+
+    public async isAppSupported() {
+        try {
+            this.transport = await this.openTransport();
+            const symbolLedger = new SymbolLedger(this.transport, 'XYM');
+            const result = await symbolLedger.isAppSupported();
+            await this.closeTransport();
+            return result;
+        } catch (error) {
+            await this.closeTransport();
+            throw this.formatError(error);
+        }
+    }
+
+    public async getAccount(path: string, networkType: number, display: boolean) {
+        try {
+            if (false === DerivationPathValidator.validate(path)) {
+                const errorMessage = 'Invalid derivation path: ' + path;
+                throw new Error(errorMessage);
+            }
+            this.transport = await this.openTransport();
+            const symbolLedger = new SymbolLedger(this.transport, 'XYM');
+            const result = await symbolLedger.getAccount(path, networkType, display);
+            await this.closeTransport();
+            return result;
+        } catch (error) {
+            await this.closeTransport();
+            throw this.formatError(error);
+        }
+    }
+
+    public async signTransaction(path: string, transferTransaction: any, networkGenerationHash: string, signerPublicKey: string) {
+        try {
+            if (false === DerivationPathValidator.validate(path)) {
```

Comment:
better to use DerivationPathValidator on left-hand-side or maybe use !DerivationPathValidator?

Reply:
Updated